queries.sql

```diff
@@ -8,4 +8,70 @@ SELECT date_of_birth FROM animals WHERE name IN ('Agumon', 'Pikachu');
 SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = TRUE;
 SELECT * FROM animals WHERE name <> 'Gabumon';
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+BEGIN TRANSACTION;
+
+UPDATE animals SET species = 'Unspecified';
+
+SELECT * FROM animals;
+
+ROLLBACK;
+
+SELECT * FROM animals;
+
+BEGIN TRANSACTION;
+
+UPDATE animals SET species = 'Digimon' WHERE name LIKE '%mon';
+
+UPDATE animals SET species = 'Pokemon' WHERE species IS NULL;
+
+SELECT * FROM animals;
+
+COMMIT;
+
+SELECT * FROM animals;
+
+BEGIN TRANSACTION;
+
+DELETE FROM animals;
+
+SELECT * FROM animals;
+
+ROLLBACK;
+
+SELECT * FROM animals;
+
+BEGIN TRANSACTION;
+
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+
+SAVEPOINT my_savepoint;
+
+UPDATE animals SET weight_kg = weight_kg * -1;
+
+ROLLBACK TO SAVEPOINT my_savepoint;
+
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+
+COMMIT;
+
+SELECT COUNT(*) AS total_animals FROM animals;
+SELECT COUNT(*) AS animals_never_escaped FROM animals WHERE escape_attempts = 0;
+
+SELECT AVG(weight_kg) AS average_weight FROM animals;
+
+SELECT neutered, SUM(escape_attempts) AS total_escape_attempts
+FROM animals
+GROUP BY neutered
+ORDER BY total_escape_attempts DESC
+LIMIT 1;
+
+SELECT species, MIN(weight_kg) AS min_weight, MAX(weight_kg) AS max_weight
+FROM animals
+GROUP BY species;
+
+SELECT species, AVG(escape_attempts) AS average_escape_attempts
+FROM animals
+WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'
+GROUP BY species;
```

Comment:
- [ ] Great work so far in crafting your database queries to align with the project requirements! 👏 However, I'd like to discuss the screenshots you've included in the pull request summary. While it's fantastic that you've documented your query results, the current format of these screenshots could use some improvement for clarity and ease of review.

  - To streamline the review process and make it more user-friendly, I recommend using your `terminal/command prompt` to execute the queries and then capture screenshots that display both the query and its results. Here's an example of what an ideal screenshot should look like:

<img width="1430" alt="Screenshot 2023-09-22 at 22 31 00" src="https://user-images.githubusercontent.com/60655304/269740895-9f4a82e0-9fd3-4a1f-bf41-3cb93fd29322.PNG">

   - This format ensures that anyone reviewing your code can quickly understand the query, its context, and the corresponding results in a single glance. It simplifies the review process and helps maintain a clear and organized documentation of your work.

Happy coding! 😊🚀🍀